contracts/modules/Swap.sol

```diff
@@ -459,6 +571,27 @@ contract Swap is BaseLogic {
         return isExactOutput ? (token1, token0) : (token0, token1);
     }
 
+    function detectAndDecodeUniV2Path(bytes memory path) private pure returns (bool, address[] memory) {
+        require(path.length >= 20 + 20, "e/swap/uni-path-length-1");
+
+        bool isUniV2 = path.length % 20 == 0;
```

Comment:
I believe a 20 hop uni3 path would also match this :)

    -> (20 + (20 * 23)) / 20
     = 24

Reply:
it would, but how probable is that 20 hop trade would even exist? the worst case scenario is that the tx would fail. is it bad considering the probability? :)